src/components/search/index.tsx

```diff
@@ -299,8 +299,11 @@ const Search: FunctionComponent<React.PropsWithChildren<SearchProps>> = ({
           indexName={TYPESENSE_COLLECTION_NAME}
           searchClient={searchClient}
           onStateChange={onSearchStateChange}
+          future={{
+            preserveSharedStateOnUnmount: true,
+          }}
```

Comment:
this is going to be default behavior for widgets soon so algolia recommends this flag